rl_bakery/operation/train_agent_operation.py

```diff
@@ -76,19 +85,18 @@ def _run(self, run_id, rl_app, engine_config, agent, timestep_df, **previous_tim
 
         # TODO: move this order and trajectory building to it's own operation so that train can
         # run on gpu while trajectory building run on spark
-        timestep_df = timestep_df.orderBy(*rl_app.env_id_cols, rl_app.ts_id_col)
+        timestep_df = timestep_df.orderBy(*self._application.env.env_id_cols, self._application.env.ts_id_col)
         traj_dict = self._trajectory_builder.run(timestep_df.collect())
 
         # setup replay buffer
         self._replay_buffer.add_batch(traj_dict)
 
         # train the agent
         global_step = tf.compat.v1.train.get_or_create_global_step()
-        summary_interval = rl_app.training_config.get("summary_interval",
-                                                      rl_app.training_config.get("log_interval", 200))
+        summary_interval = self._application.config.project.summary_interval
         with tf.compat.v2.summary.record_if(lambda: tf.math.equal(global_step % summary_interval, 0)):
-            num_iterations = rl_app.training_config["num_iterations"]
-            mini_batch_size = rl_app.training_config["mini_batch_size"]
+            num_iterations = self._application.config.training.num_iterations
```

Comment:
Accessing private members of the parent class here breaks the encapsulation. That is why the app is passed as a parameter in the method when called by the parent class.

Reply:
private is marked with two underscores, while a single underscore is 'protected', meaning derived classes have access to the values